### PR TITLE
app: BTC grant 2 (%control, %generating, %mining & %network)

### PR DESCRIPTION
**Pull request by pkova**
Here's my stuff from urbit/urbit#2005 reworked to match with your far better conventions and style.

### CODE REVIEW

File: pkg/arvo/lib/btc-node-json.hoon [hunk 1]
```diff
@@ -1816,12 +1976,286 @@
...
+      (bo res.res)
+    ::
+        %submit-block
+      :-  id.res
+      (so res.res)
+    ::
+        %submit-header
+      :-  id.res
+      (so res.res)
+    ::  Mining
```
Comment (yosoyubik) [inline pkg/arvo/lib/btc-node-json.hoon]: I guess this should be Network

File: pkg/arvo/sur/btc-node-hook.hoon [hunk 2]
```diff
@@ -1504,10 +1789,119 @@
...
+        ::
+        [%prioritise-transaction res=?]
+        ::
+        [%submit-block res=@t]
+        ::
+        [%submit-header res=@t]
+    ::  Network
+    ::
+        [%add-node res=null]
+        ::
```
Comment (yosoyubik) [inline pkg/arvo/sur/btc-node-hook.hoon]: I indented this separator comments to the left of the type declaration, following Mark's feedback from here: https://github.com/urbit/urbit/pull/1892#discussion_r366571586 I'm fine with them either way, although I'll have to change a significant amount of them from my other commits... ;)

File: pkg/arvo/sur/btc-node-hook.hoon [hunk 3]
```diff
@@ -1504,10 +1789,119 @@
...
+        [%get-peer-info (list peer-info)]
+        ::
+        [%list-banned (list banned)]
+        ::
+        [%ping res=null]
+        ::
+        [%set-ban res=null]
+        ::
+        [%set-network-active res=?]
+
```
Comment (yosoyubik) [inline pkg/arvo/sur/btc-node-hook.hoon]: Empty line

File: pkg/arvo/sur/btc-node-hook.hoon [hunk 4]
```diff
@@ -588,10 +762,121 @@
...
+        ::
+        $:  %set-ban
+            subnet=@t
+            command=?(%add %remove)
+            ban-time=(unit ?([%dr @dr] [%da @da]))
+        ==
+        ::  %setnetworkactive: Disable/enable all p2p network activity.
+        ::
+        [%set-network-active state=?]
+
```
Comment (yosoyubik) [inline pkg/arvo/sur/btc-node-hook.hoon]: Empty line

**Review by yosoyubik — COMMENTED** [review]
Thanks @pkova! Very excited to get all our work in one place so we can finally get this thing out in the world :)

I've made a few comments on your code, which I'm fine changing myself when I merge (just wanted to let you know before (: ). I was thinking of rebasing this on top of my BTC part 3 commit, just so we get the sequence of bounties in order (that means that I have to manually resolve some conflicts, basically removing the %blockchain section from part 1). 

Also, now that we consider all tags from the request type in the lib, I need to change this
`?+    -.req  ~|([%unsupported-request -.req] !!)`  to this `?-    -.req `, to avoid the `mint-vain`. I'll probably update it in the commit for part 1.

**Comment by pkova** [comment]
Fixed!

**Comment by yosoyubik** [comment]
Done! Just merged, rebased, pushed, and hopefully didn't break anything in between :D

**Pull request closed without merging**